README.md

```diff
@@ -47,6 +48,25 @@ end
 Please see the documentation for [OAuth2.Serializer](https://hexdocs.pm/oauth2/OAuth2.Serializer.html)
 for more details.
 
+## Configure a http client
+
+The http client library used is [tesla](https://github.com/elixir-tesla/tesla), the default adapter is
+hackney, to be compatible with previous versions.
+You can configure another adaptor like this:
+
+```elixir
+config :oauth2, adapter: Tesla.Adapter.Httpc
+```
+
+You can also add your own tesla middleware:
+
+```elixir
+config :oauth2, middleware: [
+  Tesla.Middleware.Retry,
+  {Tesla.Middleware.Fuse, name: :example}
+]
+```
```

Comment:
I am worried about allowing middleware for two reasons:

1. The order matters
2. If people use the wrong middleware the client may not work at all

Reply:
This is the biggest reason why I like Tesla, the middleware allow easy integration with tracing, logging and retries. The client's don't stop working with the wrong middleware but they might behave differently which is a desired effect.

Reply:
Well, yes, I 100% understand, and I also like it.

But if people put the wrong middleware they could mess up the serialization, for example. No a big deal, but worth to keep in mind.